Mimir.Worker/Scrapper/DiffScrapper.cs

```diff
@@ -57,7 +65,37 @@ public async Task ExecuteAsync(long baseIndex, long targetIndex)
         }
     }
 
-    private async void ProcessRootStateDiff(IGetDiffs_Diffs_RootStateDiff rootDiff)
+    private async Task<
+        IEnumerable<IGetTransactionSigners_Transaction_NcTransactions>
+    > GetTransactions(long processBlockIndex, long limit)
+    {
+        var operationResult = await _headlessGqlClient.GetTransactionSigners.ExecuteAsync(
+            processBlockIndex,
+            limit
+        );
+
+        if (
+            operationResult.Data?.Transaction?.NcTransactions == null
+            || !operationResult.Data.Transaction.NcTransactions.Any()
+        )
+        {
+            Serilog.Log.Error(
+                "No transactions found or null data. Process Block Index: {ProcessBlockIndex}",
+                processBlockIndex
+            );
+            return Enumerable.Empty<IGetTransactionSigners_Transaction_NcTransactions>();
+        }
+
+        return operationResult
+            .Data.Transaction.NcTransactions.Where(tx => tx != null)
+            .Cast<IGetTransactionSigners_Transaction_NcTransactions>()
+            .ToList();
```

Comment:
```suggestion
        return operationResult
            .Data.Transaction.NcTransactions
            .OfType<IGetTransactionSigners_Transaction_NcTransactions>()
            .ToList();
```

I tested with the below code:

```csharp
using System;
using System.Linq;
using System.Collections.Generic;

interface IApp {
	
}

class App : IApp {
}

public class Program
{
	public static void Main()
	{
		var list = new List<IApp> { new App(), null };
		var newList = list.OfType<IApp>().ToList();
		Console.WriteLine(list.Count);  // 2
		Console.WriteLine(newList.Count);  // 1
	}
}
```

Reply:
Good, awesome method